### PR TITLE
AI21: tools calling support in Langchain

### PR DESCRIPTION
**Pull request by amirai21**
This pull request introduces support for the AI21 tools calling feature, available by the Jamba-1.5 models. When Jamba-1.5 detects the necessity to invoke a provided tool, as indicated by the 'tools' parameter passed to the model:

```
class ToolDefinition(TypedDict, total=False):
    type: Required[Literal["function"]]
    function: Required[FunctionToolDefinition]

class FunctionToolDefinition(TypedDict, total=False):
    name: Required[str]
    description: str
    parameters: ToolParameters

class ToolParameters(TypedDict, total=False):
    type: Literal["object"]
    properties: Required[Dict[str, Any]]
    required: List[str]
```

It will respond with a list of tool calls structured as follows:

```
class ToolCall(AI21BaseModel):
    id: str
    function: ToolFunction
    type: Literal["function"] = "function"

class ToolFunction(AI21BaseModel):
    name: str
    arguments: str
```

This pull request incorporates the necessary modifications to integrate this functionality into the ai21-langchain library.

### CODE REVIEW

File: libs/partners/ai21/langchain_ai21/chat_models.py [hunk 1]
```diff
@@ -48,14 +63,14 @@ class ChatAI21(BaseChatModel, AI21Base):
...
-    max_tokens: int = 16
+    max_tokens: int = 512
     """The maximum number of tokens to generate for each response."""
 
     min_tokens: int = 0
     """The minimum number of tokens to generate for each response.
     _Not supported for all models._"""
 
-    temperature: float = 0.7
+    temperature: float = 0.4
```
Comment (baskaryan) [inline libs/partners/ai21/langchain_ai21/chat_models.py:73]: is this related to this pr?

Comment (pazshalev) [reply libs/partners/ai21/langchain_ai21/chat_models.py:73]: Hi @baskaryan, this isn't related directly to this pr, we've just decided internally to set the default temperature to 0.4 instead of 0.7.

**Review by efriis — APPROVED** [review]
